[Table] Introduce padding property

eyn opened the pull request:
- Introduce TableCell padding property.
- Remove unused code

### Breaking changes

```diff
-          <TableCell checkbox>
+          <TableCell padding="checkbox">
```

oliviertassinari commented on src/Table/TableCell.js:
```diff
@@ -4,24 +4,22 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import type { ElementType, Node } from 'react';
 import classNames from 'classnames';
+import deprecated from 'prop-types-extra/lib/deprecated';
```
Please, let's avoid the depreciation. Introduction breaking changes save maintainer times.

eyn commented:
I originally wanted to minimise changes but have also renamed compact -> dense across the class. Removed deprecations etc. Do you want api doc updates as part of PRs or do you do them seperately?

oliviertassinari commented:
@eyn Thanks
